apps/sumologic/src/app/install/page.tsx

```diff
@@ -0,0 +1,214 @@
+'use client';
+
+import React, { useState, useEffect } from 'react';
+import { useFormState } from 'react-dom';
+import { useSearchParams } from 'next/navigation';
+import styles from '../styles.module.css';
+import { install } from './action';
+import type { FormState } from './action';
+
+function Step({
+  number,
+  text,
+  onClick,
+  active,
+}: {
+  number: string;
+  text: string;
+  active: boolean;
+  onClick: () => void;
+}) {
+  return (
+    <div className={styles.step}>
+      <button onClick={onClick} style={{ border: '0', background: 'transparent' }} type="button">
+        <span
+          className={styles.step_number}
+          style={{
+            backgroundColor: active ? '#22bb33' : 'gainsboro',
+            color: active ? 'white' : 'black',
+            fontWeight: active ? 'bold' : 'normal',
+          }}>
+          {number}
+        </span>
+      </button>
+      <span className={styles.step_text} style={{ fontWeight: active ? 'bold' : 'normal' }}>
+        {text}
+      </span>
+    </div>
+  );
+}
+
+function InstructionItems({
+  heading,
+  instructions,
+  active,
+}: {
+  heading: string;
+  instructions: string[];
+  active: string;
+}) {
+  return (
+    <div className={styles.instructions_container}>
+      <h1>{heading}</h1>
+      {instructions.map((item, index) => (
+        <div className={styles.instruction} key={item}>
+          {' '}
+          {/* Changed key to index for uniqueness */}
+          <span className={styles.instructionNumber}>{index + 1}</span>
+          <span className={styles.instructionText}>
+            {index === 0 && active === '1' ? (
+              <span>
+                In{' '}
+                <a
+                  href="https://www.sumologic.com/sign-up/"
+                  rel="noopener noreferrer"
+                  target="_blank">
+                  Sumo Logic
+                </a>
+                {item}.
+              </span>
+            ) : (
+              item
+            )}
+          </span>
+        </div>
+      ))}
+    </div>
+  );
+}
+
+function InstructionsModal() {
+  const [active, setActive] = useState<string>('1');
+  const searchParams = useSearchParams();
+  const organisationId = searchParams.get('organisation_id');
+  const region = searchParams.get('region');
+
+  const [state, formAction] = useFormState<FormState, FormData>(install, {});
+  const [selectedDomain, setSelectedDomain] = useState('');
+
+  const domainOptions = {
+    AU: 'https://api.au.sumologic.com/api/',
+    CA: 'https://api.ca.sumologic.com/api/',
+    DE: 'https://api.de.sumologic.com/api/',
+    EU: 'https://api.eu.sumologic.com/api/',
+    FED: 'https://api.fed.sumologic.com/api/',
+    IN: 'https://api.in.sumologic.com/api/',
+    JP: 'https://api.jp.sumologic.com/api/',
+    US1: 'https://api.us1.sumologic.com/api/',
+    US2: 'https://api.us2.sumologic.com/api/',
+  };
```

Comment:
Let's exports this from `action.tsx`, so we can share it in the zod schema + component
```suggestion
export  const sourceRegions = [
    'au',
    'ca',
    'de',
    'eu',
    'fed',
    'in',
    'jp',
    'us1',
    'us2'
]
```